src/quest_manager/views.py

```diff
@@ -635,6 +635,43 @@ def detail(request, quest_id):
 #
 # #################################
 
+@method_decorator(staff_member_required, name='dispatch')
+class Ajax_Approve(NonPublicOnlyViewMixin, View):
+    """ alternate approve view for quick reply approves.
+    Should wrap "approve" for consistency.
```

Comment:
Why not just extend the Approve view?  or can you just steal some of the approve view's methods to prevent duplication?  If what have in two spots without a cross reference, this will cause problems when changing in the future.

Maybe that's not feasible, just wondering why you put the note here instead of just doing it =)

Reply:
i forgot why i didnt (been a month), might be because I found it hard to read. Ended up refactoring the approve code from a fbv to a cbv that can handle ajax requests